src/game/Unit.cpp

```diff
@@ -2847,12 +2937,19 @@ SpellMissInfo Unit::MagicSpellHitResult(Unit* pVictim, SpellEntry const* spell)
 //   Resist
 SpellMissInfo Unit::SpellHitResult(Unit* pVictim, SpellEntry const* spell, bool CanReflect)
 {
+    SpellSchoolMask schoolMask = GetSpellSchoolMask(spell);
+
+    // wand case
+    bool wand = spell->Id == 5019;
+    if (wand && !!(getClassMask() & CLASSMASK_WAND_USERS) && GetTypeId() == TYPEID_PLAYER)
+        schoolMask = SpellSchoolMask(1 <<GetWeaponDamageSchool(RANGED_ATTACK));
```

Comment:
Same conversion here, however it was introduced in the wand commit and this commit seems to work pretty well...